src/docs/development/tools/sdk/upgrading.md

```diff
@@ -29,36 +26,85 @@ dev_dependencies:
     sdk: flutter
 ```
 
-The `sdk: flutter` line tells the `flutter` command-line tool to find the
-correct package for you.
-
-Do not use the `pub get` or `pub upgrade` commands to manage your dependencies.
+{{site.alert.warning}}
+Don't use the `pub get` or `pub upgrade` commands to manage dependencies
+for Flutter apps.
 Instead, use `flutter packages get` or `flutter packages upgrade`.
 If you want to use pub manually, you can run it directly by setting the
 `FLUTTER_ROOT` environment variable.
+{{site.alert.end}}
+
 
-## Upgrading Flutter channel and your packages
+## Upgrading the Flutter SDK and packages
 
-To update both the Flutter SDK and your packages, use the `flutter upgrade`
-command from the root of your app (the same directory that contains the
-`pubspec.yaml` file):
+To update both the Flutter SDK and the packages that your app depends on,
+use the `flutter upgrade` command from the root of your app
+(the same directory that contains the `pubspec.yaml` file):
 
 ```terminal
 $ flutter upgrade
 ```
 
-## Upgrading your packages
+This command first gets the most recent version of the Flutter SDK
+that's available on your Flutter channel.
+Then this command updates each package that your app depends on
+to the most recent compatible version.
+
+If you want an even more recent version of the Flutter SDK,
+switch to a less stable Flutter channel
+and then run `flutter upgrade`.
+
+### Switching your Flutter channel
+
+Flutter has [four release channels][Flutter release channel]:
+**stable**, **beta**, **dev**, and **master**.
+We recommend using the **{{site.sdk.channel}}** channel
+unless you need a more recent release.
+
+To view your current channel, use `flutter channel`.
+To change to another channel, use `flutter channel <channel-name>`.
+Once you've changed your channel, use `flutter upgrade`
+to download the Flutter SDK and dependent packages.
+
+```terminal
+$ flutter channel     # Shows your current channel
+$ flutter channel dev # Changes your channel to the dev channel
+$ flutter upgrade     # Get the Flutter SDK and dependent packages
+```
```

Comment:
I'm not a fan of having comments in the `terminal` code blocks because it makes it a bit more difficult to copy the terminal command. (Also, it makes for duplication -- similar text appears in the prose and terminal blocks.)

How about something like the following?

---

To view your current channel, use the following command:

 ```terminal
$ flutter channel
```

To change to another channel, use `flutter channel <channel-name>`. Once you've changed your channel, use `flutter upgrade` to download the Flutter SDK and dependent packages. For example:

 ```terminal
$ flutter channel dev
$ flutter upgrade
```